packages/algob/src/internal/deployer.ts

```diff
@@ -573,6 +574,47 @@ export class DeployerDeployMode extends DeployerBasicMode implements Deployer {
 		persistCheckpoint(this.txWriter.scriptName, this.cpData.strippedCP);
 	}
 
+	/**
+	 * Save new ASA information(get from txConfirmation) to checkpoint
+	 * @param asaName name of asa
+	 * @param txConfirmation txn confirmation
+	 */
+	addCheckpointASA(asaName: string, txConfirmation: ConfirmedTxInfo): void {
+		const txn = txConfirmation.txn.txn;
+
+		this.cpData.registerASA(this.networkName, asaName, {
+			assetIndex: txConfirmation["asset-index"],
+			creator: algosdk.encodeAddress(txn.apar?.m as Uint8Array),
+			confirmedRound: txConfirmation["confirmed-round"],
+			deleted: false,
+			assetDef: {
+				total: txn.apar?.t as number,
+				decimals: txn.apar?.dc as number,
+			},
+			txID: algosdk.Transaction.from_obj_for_encoding(txn).txID(),
+		});
+	}
+
+	/**
+	 * Save new application information(get from txConfirmation) to checkpoint
+	 * @param appName app name
+	 * @param txConfirmation txn confirmation
+	 */
+	addCheckpointApp(appName: string, txConfirmation: ConfirmedTxInfo): void {
+		const { txn } = txConfirmation.txn;
+		this.cpData.registerSSC(this.networkName, appName, {
+			appID: txConfirmation["application-index"],
+			applicationAccount: algosdk.getApplicationAddress(txConfirmation["application-index"]),
```

Comment:
+ Confused here. How do we ensure here that the confirmed transaction is an app call/app deploy? Or if we're calling this specifically "after" that app deploy call -- we should rename this function to `addInnerAppDeployCP`/`addInnerAppDeployCheckpoint`

+ Is it possible to deploy multiple apps during inner transaction? In that case is this function called multiple times?